### PR TITLE
resolving PrometheusExportService path comparison failure

### LINKED ISSUE #3731: Prometheus metrics route never matches in 1.0.0-M4
In `1.0.0-M4` and `master` at this time, the `pathInfo` field of `Request` was changed to a `Uri`, however the Prometheus `/metrics` route still does an equality check against a string which simply never passes.

https://github.com/http4s/http4s/blob/v1.0.0-M4/prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/PrometheusExportService.scala#L49

### PR DESCRIPTION
**Pull request by jyoo980**
Should be work that resolves #3731

### CODE REVIEW

File: prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/PrometheusExportService.scala [hunk 1]
```diff
@@ -46,7 +46,7 @@ object PrometheusExportService {
 
   def service[F[_]: Sync](collectorRegistry: CollectorRegistry): HttpRoutes[F] =
     HttpRoutes.of[F] {
-      case req if req.method == Method.GET && req.pathInfo == "/metrics" =>
+      case req if req.method == Method.GET && req.pathInfo.toString == "/metrics" =>
```
Comment (jyoo980) [inline prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/PrometheusExportService.scala:49]: Alternatively, we could do
```Scala
req.pathInfo == path"/metrics"
```
which is something I've seen used in the codebase before.

Comment (rossabaker) [reply prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/PrometheusExportService.scala:49]: I like the `path` version, and we could store that to a constant in the service.

And if we used cats' `===`, the bug wouldn't have compiled.

**Comment by rossabaker** [comment]
Thanks! When this is ready, we'll cherrypick it back to series/0.21.

File: prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/PrometheusExportService.scala (hunk 1 above)
Comment (jyoo980) [reply prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/PrometheusExportService.scala]: Done!

**Comment by jyoo980** [comment]
Looks like this was the only place where we were doing a raw string comparison for the path; other usages had the correct comparison to a `Path`

**Review by rossabaker — APPROVED** [review]
Thanks!

**Review by hamnis — APPROVED** [review]
Apparently I missed a spot.
The universal equality things are sometimes hard to find.

Good find. thanks for the fix

**Comment by hamnis** [comment]
> Thanks! When this is ready, we'll cherrypick it back to series/0.21.

Why? this is only a problem for 1.0 and master

**Comment by rossabaker** [comment]
You're right.  I was looking for binary compatibility without thinking.

**Pull request merged**